scico/test/linop/test_linop.py

```diff
@@ -559,3 +559,39 @@ def test_sum_bad_shapes(sumtestobj, axis):
     x = sumtestobj.x
     with pytest.raises(ValueError):
         A = linop.Sum(input_shape=x.shape, input_dtype=x.dtype, sum_axis=axis)
+
+
+class SliceTestObj:
+    def __init__(self, dtype):
+        self.x = snp.zeros((4, 5, 6, 7), dtype=dtype)
+
+
+@pytest.fixture(scope="module", params=[np.float32, np.complex64])
+def slicetestobj(request):
+    yield SliceTestObj(request.param)
+
+
+slice_examples = [
+    np.s_[1:],
+    np.s_[:, 2:],
+    np.s_[..., 3:],
+    np.s_[1:, :-3],
+    np.s_[1:, :, :3],
+    np.s_[1:, ..., 2:],
+]
+
+
+@pytest.mark.parametrize("slc", slice_examples)
+def test_slice_eval(slicetestobj, slc):
+    x = slicetestobj.x
```

Comment:
Possible alternative: remove `SliceTestObj` and `slicetestobj`, replace line 586 with
`x = snp.zeros((4, 5, 6, 7), dtype=dtype)`, parametrize over `dtype`.

My eyes glaze over when I see fixtures, and what is it gaining us here? Perhaps it caches `x`?

Reply:
I feel the same way. The approach here is a direct copy of the tests for the `Sum` operator. No arguments on simplifying the new tests, but it would seem to make sense to simplify the ones for `Sum` at the same time?

Reply:
Agreed. I made an issue for fixing it, let's leave it as is for this PR.